### PR TITLE
Add Rust client docs, Rust syntax highlighting

### PR DESCRIPTION
**Pull request by slvrtrn**
## Summary

Most of the things from https://github.com/ClickHouse/clickhouse-rs/tree/main?tab=readme-ov-file#clickhouse-rs were moved to the docs repo

I added several new sections
* [Paul Loyd](https://github.com/loyd) recognition as the original author 
* An explicit link to the examples
* HTTPS/CH Cloud connection
* Async insert example
* Slightly adjusted `inserter` example (with `end` call mentioned in the code as it caused confusion)
* DDLs with `wait_end_of_query`
* CH settings usage examples
* QueryID, SessionID examples
* Custom HTTP headers example
* Custom HTTP client example
* `CANNOT_READ_ALL_DATA` troubleshooting
* "Contact us"

Leftovers for the follow-ups:
- Rework "Data Types" section, especially when the data types examples PR is merged
- Add more docs based on the current/past issues in the client repo

## Checklist
- [x] If adding a new integration page, also add an entry to the integrations list here: https://github.com/ClickHouse/clickhouse-docs/blob/main/docs/en/integrations/index.mdx

### CODE REVIEW

File: docs/en/integrations/language-clients/rust.md [hunk 1]
```diff
@@ -0,0 +1,517 @@
+---
+sidebar_label: Rust
+sidebar_position: 4
+keywords: [clickhouse, rs, rust, http, client, connect, integrate]
```
Comment (mshustov) [inline docs/en/integrations/language-clients/rust.md:4]: cargo, crate?

File: docs/en/integrations/language-clients/rust.md [hunk 2]
```diff
@@ -0,0 +1,517 @@
...
+:::important
+When connecting to ClickHouse via an `HTTPS` url, either the `native-tls` or `rustls-tls` feature should be enabled.
+If both are enabled, the `rustls-tls` feature will take precedence.
+:::
+
+## ClickHouse versions compatibility
+
+The client is compatible with the LTS or newer versions of ClickHouse, as well as ClickHouse Cloud.
+
+ClickHouse server older than v22.6 handles RowBinary [incorrectly in some rare cases](https://github.com/ClickHouse/ClickHouse/issues/37420).
```
Comment (mshustov) [inline docs/en/integrations/language-clients/rust.md:58]: Let's mention the version we test against: It should be a maintained version. As of today, it's `24.3`, and `24.6+`

File: docs/en/integrations/language-clients/rust.md [hunk 3]
```diff
@@ -0,0 +1,517 @@
...
+let client = Client::default()
+    // should include both protocol and port
+    .with_url("http://localhost:8123")
+    .with_user("name")
+    .with_password("123")
+    .with_database("test");
+```
+
+:::tip
+Reuse created clients or clone them in order to reuse the underlying hyper connection pool.
```
Comment (mshustov) [inline docs/en/integrations/language-clients/rust.md:87]: optional: move to the top, it's easy to miss

File: docs/en/integrations/language-clients/rust.md [hunk 4]
```diff
@@ -0,0 +1,517 @@
...
+    env::var(key).unwrap_or_else(|_| panic!("{key} env variable should be set"))
+}
+
+let client = Client::default()
+    .with_url(read_env_var("CLICKHOUSE_URL"))
+    .with_user(read_env_var("CLICKHOUSE_USER"))
+    .with_password(read_env_var("CLICKHOUSE_PASSWORD"));
+```
+
+NB: the URL should include both protocol and port, e.g. `https://instance.clickhouse.cloud:8443`.
```
Comment (mshustov) [inline docs/en/integrations/language-clients/rust.md:107]: optional: it's a potential source of many misconfigurations, so it should be `important` or `warning`, not just NB

File: docs/en/integrations/language-clients/rust.md [hunk 5]
```diff
@@ -0,0 +1,517 @@
...
+
+```rust
+let client = Client::default()
+    .with_url("http://localhost:8123")
+    .with_option("async_insert", "1")
+    .with_option("wait_for_async_insert", "0");
+```
+
+See also:
+- [Async insert example](https://github.com/ClickHouse/clickhouse-rs/blob/main/examples/async_insert.rs) in the client repo.
```
Comment (mshustov) [inline docs/en/integrations/language-clients/rust.md:181]: ❤️ this runnable examples

File: docs/en/integrations/language-clients/rust.md [hunk 6]
```diff
@@ -0,0 +1,517 @@
...
+let client = Client::with_http_client(hyper_client).with_url("http://localhost:8123");
+```
+
+:::warning
+This example relies on the legacy Hyper API and is a subject to change in the future.
+:::
+
+See also: [custom HTTP client example](https://github.com/ClickHouse/clickhouse-rs/blob/main/examples/custom_http_client.rs) in the client repo.
+
+## Data Types
```
Comment (mshustov) [inline docs/en/integrations/language-clients/rust.md:332]: should we add a table with supported type, as you did in https://clickhouse.com/docs/en/integrations/javascript#supported-clickhouse-data-types ?

File: docs/en/integrations/language-clients/rust.md [hunk 7]
```diff
@@ -0,0 +1,517 @@
...
+In this example, this is fixed by the correct definition of the `EventLog` struct:
+
+```rust
+#[derive(Debug, Serialize, Deserialize, Row)]
+struct EventLog {
+    id: u32
+}
+```
+
+## Contact us
```
Comment (mshustov) [inline docs/en/integrations/language-clients/rust.md:515]: Should we add known limitations? Like this https://github.com/ClickHouse/clickhouse-rs/issues/74

File: docs/en/integrations/language-clients/rust.md [hunk 8]
```diff
@@ -0,0 +1,517 @@
...
+
+#[derive(Row, Deserialize)]
+struct MyRow<'a> {
+    no: u32,
+    name: &'a str,
+}
+
+let table_name = "some";
+let mut cursor = client
+    .query("SELECT ?fields FROM ? WHERE no BETWEEN ? AND ?")
```
Comment (mshustov) [inline docs/en/integrations/language-clients/rust.md:127]: Does the client support server-side bindings?

**Review by mshustov — APPROVED** [review]

File: docs/en/integrations/language-clients/rust.md (hunk 6 above)
Comment (slvrtrn) [reply docs/en/integrations/language-clients/rust.md:332]: Yes, that is ideally a part of the rework in the follow-ups.

File: docs/en/integrations/language-clients/rust.md (hunk 8 above)
Comment (slvrtrn) [reply docs/en/integrations/language-clients/rust.md:127]: Not as of now.

Comment (slvrtrn) [reply docs/en/integrations/language-clients/rust.md:127]: https://github.com/ClickHouse/clickhouse-rs/issues/142

File: docs/en/integrations/language-clients/rust.md (hunk 2 above)
Comment (slvrtrn) [reply docs/en/integrations/language-clients/rust.md:58]: It will be added later in a follow-up when we change our test matrix.

**Pull request merged**